Cannot build encrypted images on Zephyr

Building with rsa-oaep:

```
west build -b nrf52840dk_nrf52840 -d build_boot bootloader/mcuboot/boot/zephyr -- -DCONFIG_BOOT_ENCRYPT_RSA=y
```
results in:

```
...
In file included from /home/utzig/src/Nordic/mcuboot/boot/bootutil/zephyr/../include/bootutil/enc_key.h:33,
                 from /home/utzig/src/Nordic/mcuboot/boot/bootutil/src/bootutil_priv.h:43,
                 from /home/utzig/src/Nordic/mcuboot/boot/bootutil/src/bootutil_public.c:48:
/home/utzig/src/Nordic/mcuboot/boot/bootutil/zephyr/../include/bootutil/crypto/aes_ctr.h:23:14: fatal error: mbedtls/aes.h: No such file or directory
   23 |     #include <mbedtls/aes.h>
      |              ^~~~~~~~~~~~~~~
compilation terminated.
```

Building with ec256:

```
west build -b nrf52840dk_nrf52840 -d build_boot bootloader/mcuboot/boot/zephyr -- -DCONFIG_BOOT_SIGNATURE_TYPE_ECDSA_P256=y -DCONFIG_BOOT_ENCRYPT_EC256=y
```
results in:

```
...
In file included from /home/utzig/src/Nordic/mcuboot/boot/bootutil/zephyr/../include/bootutil/enc_key.h:33,
                 from /home/utzig/src/Nordic/mcuboot/boot/bootutil/src/bootutil_priv.h:43,
                 from /home/utzig/src/Nordic/mcuboot/boot/bootutil/src/bootutil_public.c:48:
/home/utzig/src/Nordic/mcuboot/boot/bootutil/zephyr/../include/bootutil/crypto/aes_ctr.h:30:14: fatal error: tinycrypt/aes.h: No such file or directory
   30 |     #include <tinycrypt/aes.h>
      |              ^~~~~~~~~~~~~~~~~
compilation terminated.
```

This fails since bootutil was separated into a public part. I belive it stems from the bootutil_public.c CMakeLists.txt not adding the proper include paths, which is only required by bootutil_priv.h. Eg, I was able to fix EC256 with the following diff:

```diff
--- a/boot/bootutil/zephyr/CMakeLists.txt
+++ b/boot/bootutil/zephyr/CMakeLists.txt
@@ -12,6 +12,10 @@ target_include_directories(MCUBOOT_BOOTUTIL INTERFACE
   ../../zephyr/include
 )
 
+if(CONFIG_BOOT_ENCRYPT_EC256)
+  zephyr_include_directories(../../../ext/tinycrypt/lib/include)
+endif()
+
 zephyr_library()
 zephyr_library_sources(
   ../src/bootutil_public.c
```

--- a/boot/bootutil/include/bootutil/bootutil_public.h
+++ b/boot/bootutil/include/bootutil/bootutil_public.h
@@ -88,6 +88,36 @@ extern "C" {
 
 #define BOOT_MAGIC_SZ (sizeof boot_img_magic)
 
+#define BOOT_EFLASH      1
+#define BOOT_EFILE       2
+#define BOOT_EBADIMAGE   3
+#define BOOT_EBADVECT    4
+#define BOOT_EBADSTATUS  5
+#define BOOT_ENOMEM      6
+#define BOOT_EBADARGS    7
+#define BOOT_EBADVERSION 8
+
+/*
+ * Extract the swap type and image number from image trailers's swap_info
+ * filed.
+ */
+#define BOOT_GET_SWAP_TYPE(swap_info)    ((swap_info) & 0x0F)
+#define BOOT_GET_IMAGE_NUM(swap_info)    ((swap_info) >> 4)
+
+/* Construct the swap_info field from swap type and image number */
+#define BOOT_SET_SWAP_INFO(swap_info, image, type)  {                          \
+                                                    assert((image) < 0xF);     \
+                                                    assert((type)  < 0xF);     \
+                                                    (swap_info) = (image) << 4 \
+                                                                | (type);      \
+                                                    }
+#ifdef MCUBOOT_HAVE_ASSERT_H
+#include "mcuboot_config/mcuboot_assert.h"
+#else
+#include <assert.h>
+#define ASSERT assert
+#endif
+
 struct boot_swap_state {
     uint8_t magic;      /* One of the BOOT_MAGIC_[...] values. */
     uint8_t swap_type;  /* One of the BOOT_SWAP_TYPE_[...] values. */

--- a/boot/bootutil/include/bootutil/enc_key.h
+++ b/boot/bootutil/include/bootutil/enc_key.h
@@ -32,28 +32,13 @@
 #include <flash_map_backend/flash_map_backend.h>
 #include "bootutil/crypto/aes_ctr.h"
 #include "bootutil/image.h"
+#include "bootutil/enc_key_public.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#define BOOT_ENC_KEY_SIZE       16
 #define BOOT_ENC_KEY_SIZE_BITS  (BOOT_ENC_KEY_SIZE * 8)
-
-#define TLV_ENC_RSA_SZ    256
-#define TLV_ENC_KW_SZ     24
-#define TLV_ENC_EC256_SZ  (65 + 32 + 16)
-#define TLV_ENC_X25519_SZ (32 + 32 + 16)
-
-#if defined(MCUBOOT_ENCRYPT_RSA)
-#define BOOT_ENC_TLV_SIZE TLV_ENC_RSA_SZ
-#elif defined(MCUBOOT_ENCRYPT_EC256)
-#define BOOT_ENC_TLV_SIZE TLV_ENC_EC256_SZ
-#elif defined(MCUBOOT_ENCRYPT_X25519)
-#define BOOT_ENC_TLV_SIZE TLV_ENC_X25519_SZ
-#else
-#define BOOT_ENC_TLV_SIZE TLV_ENC_KW_SZ
-#endif
 
 #define BOOT_ENC_TLV_ALIGN_SIZE \
     ((((BOOT_ENC_TLV_SIZE - 1) / BOOT_MAX_ALIGN) + 1) * BOOT_MAX_ALIGN)

--- a/boot/bootutil/include/bootutil/enc_key_public.h
+++ b/boot/bootutil/include/bootutil/enc_key_public.h
@@ -1,0 +1,56 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2018-2019 JUUL Labs
+ * Copyright (c) 2019 Arm Limited
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * Original license:
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef BOOTUTIL_ENC_KEY_PUBLIC_H
+#define BOOTUTIL_ENC_KEY_PUBLIC_H
+#include <mcuboot_config/mcuboot_config.h>
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define BOOT_ENC_KEY_SIZE       16
+
+#define TLV_ENC_RSA_SZ    256
+#define TLV_ENC_KW_SZ     24
+#define TLV_ENC_EC256_SZ  (65 + 32 + 16)
+#define TLV_ENC_X25519_SZ (32 + 32 + 16)
+
+#if defined(MCUBOOT_ENCRYPT_RSA)
+#define BOOT_ENC_TLV_SIZE TLV_ENC_RSA_SZ
+#elif defined(MCUBOOT_ENCRYPT_EC256)
+#define BOOT_ENC_TLV_SIZE TLV_ENC_EC256_SZ
+#elif defined(MCUBOOT_ENCRYPT_X25519)
+#define BOOT_ENC_TLV_SIZE TLV_ENC_X25519_SZ
+#else
+#define BOOT_ENC_TLV_SIZE TLV_ENC_KW_SZ
+#endif
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* BOOTUTIL_ENC_KEY_PUBLIC_H */

--- a/boot/bootutil/src/bootutil_priv.h
+++ b/boot/bootutil/src/bootutil_priv.h
@@ -47,23 +47,7 @@
 extern "C" {
 #endif
 
-#ifdef MCUBOOT_HAVE_ASSERT_H
-#include "mcuboot_config/mcuboot_assert.h"
-#else
-#include <assert.h>
-#define ASSERT assert
-#endif
-
 struct flash_area;
-
-#define BOOT_EFLASH      1
-#define BOOT_EFILE       2
-#define BOOT_EBADIMAGE   3
-#define BOOT_EBADVECT    4
-#define BOOT_EBADSTATUS  5
-#define BOOT_ENOMEM      6
-#define BOOT_EBADARGS    7
-#define BOOT_EBADVERSION 8
 
 #define BOOT_TMPBUF_SZ  256
 
@@ -168,21 +152,6 @@ _Static_assert(BOOT_IMAGE_NUMBER > 0, "Invalid value for BOOT_IMAGE_NUMBER");
 #endif /* MCUBOOT_DIRECT_XIP || MCUBOOT_RAM_LOAD */
 
 #define BOOT_MAX_IMG_SECTORS       MCUBOOT_MAX_IMG_SECTORS
-
-/*
- * Extract the swap type and image number from image trailers's swap_info
- * filed.
- */
-#define BOOT_GET_SWAP_TYPE(swap_info)    ((swap_info) & 0x0F)
-#define BOOT_GET_IMAGE_NUM(swap_info)    ((swap_info) >> 4)
-
-/* Construct the swap_info field from swap type and image number */
-#define BOOT_SET_SWAP_INFO(swap_info, image, type)  {                          \
-                                                    assert((image) < 0xF);     \
-                                                    assert((type)  < 0xF);     \
-                                                    (swap_info) = (image) << 4 \
-                                                                | (type);      \
-                                                    }
 
 #define BOOT_LOG_IMAGE_INFO(slot, hdr)                                    \
     BOOT_LOG_INF("%-9s slot: version=%u.%u.%u+%u",                        \

--- a/boot/bootutil/src/bootutil_public.c
+++ b/boot/bootutil/src/bootutil_public.c
@@ -47,7 +47,7 @@
 #include "bootutil/bootutil_public.h"
 #include "bootutil/bootutil_log.h"
 #ifdef MCUBOOT_ENC_IMAGES
-#include "bootutil/enc_key.h"
+#include "bootutil/enc_key_public.h"
 #endif
 
 #ifdef CONFIG_MCUBOOT

--- a/boot/bootutil/src/bootutil_public.c
+++ b/boot/bootutil/src/bootutil_public.c
@@ -45,7 +45,6 @@
 
 #include "bootutil/image.h"
 #include "bootutil/bootutil_public.h"
-#include "bootutil_priv.h"
 #include "bootutil/bootutil_log.h"
 #ifdef MCUBOOT_ENC_IMAGES
 #include "bootutil/enc_key.h"

